### ISSUE TITLE
Do not generate attributes with value `undefined`

### ISSUE THREAD

**Issue by edzis** [nashwaan/xml-js#50]
I use some optional attributes that I want to render only when they are set. 
When some of them are undefined, current xml output is `myAttr="undefined"`, which makes the document not valid.
My current workaround is to use `lodash` with

    {
      _arguments: pickBy({
		someProp: 'fixed',
		otherProp1: optionalValue1,
		...
		otherPropN: optionalValueN,
	  })
	}

**Comment by nashwaan** [comment]
@edzis 
I am not sure that undefined attributes should be completely skipped. Maybe your suggestion can be implemented as an option so that existing behavior will not be impacted.

**Comment by edzis** [comment]
@nashwaan thus makes sense to me

**Comment by nashwaan** [comment]
Mind to arrange a pull request for this?

**Comment by edzis** [comment]
Will explore the code to gather some ideas on how to implement this by early next week.

**Comment by cminini** [comment]
As of today, is there any way to not render optional/empty attributes? thanks

**Comment by edzis** [comment]
@cminini Either filter the keys as I mention in description or wait for #70 to be merged

**Comment by nashwaan** [comment]
@edzis Thanks for your contribution. This is now published in `v1.6.6`

**Comment by TChengZ** [comment]
how to not generate element with value 'undefined' i can not find any option to support this
```
<childs>
            <layerObjectJS>
                <type>Text</type>
            </layerObjectJS>
            <layerObject/>
            <objectID>07E882F4-9C63-4038-A807-144C9C79247A</objectID>
            <name>Type something</name>
        </childs>
```
i dont want the element of layerObject and i dont want the element of layerObjectJS althought it has value
what i want is 
```
<childs>
            <objectID>07E882F4-9C63-4038-A807-144C9C79247A</objectID>
            <name>Type something</name>
        </childs>
```
